sharktank/sharktank/examples/export_paged_llm_v1.py

```diff
@@ -153,33 +153,33 @@ def setup_cache(model, shard_count):
             )
             page_dim = torch.export.Dim("page")
 
-            dynamic_shapes = [
-                {0: page_dim} for _ in range(llama_config.pipeline_parallelism_size)
-            ]
+            pipeline_parallelism_size = len(cache_state)
+            tensor_parallelism_size = (
+                cache_state[0].shard_count
+                if isinstance(cache_state[0], ShardedTensor)
+                else 1
+            )
+
+            dynamic_shapes = [{0: page_dim} for _ in range(pipeline_parallelism_size)]
             unpacked = cache_state
             arg_affinities = {}
             shard_dim = None
 
             # Need to unpack that state when sharded (for tracing support reasons)
-            if (
-                llama_config.tensor_parallelism_size > 1
-                or llama_config.pipeline_parallelism_size > 1
-            ):
+            if tensor_parallelism_size > 1 or pipeline_parallelism_size > 1:
                 shard_dim = cache_state[0].shard_dim
 
                 unpacked = [[shard._data for shard in cs.shards] for cs in cache_state]
                 dynamic_shapes = [
-                    [ds] * llama_config.tensor_parallelism_size for ds in dynamic_shapes
+                    [ds] * tensor_parallelism_size for ds in dynamic_shapes
```

Comment:
Move this to where `dynamic_shapes` is constructed. Doing it after the fact feels wrong.